File: Leetcode/53. Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(int A[], int n) {
        int i,j,k,ans,cs;
        ans=cs=A[0];
        for(i=1;i<n;i++)
        {
            if(cs<0)
            cs=0;
            cs+=A[i];
            ans=max(ans,cs);
        }
        return ans;
    }
};

File: Leetcode/53. Maximum Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int n = nums.length;
        if(n==1){
            return nums[0];
        }
        int ans = nums[0];
        int currMax = nums[0];
        for(int i=1;i<n;i++){
            if(currMax + nums[i] >= nums[i]){
                currMax = currMax + nums[i];
            }else{
                currMax = nums[i];
            }
            ans = Math.max(ans, currMax);
        }
        return ans;
    }
}
